cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -306,6 +306,11 @@ public void kill(ProgramRunInfo programRunInfo) throws Exception {
     stop(programRunInfo);
   }
 
+  @Override
+  public void kill(ProgramRunInfo programRunInfo, RuntimeJobStatus status) throws Exception {
+    stop(programRunInfo);
```

Comment:
this shouldn't call the existing stop(), as that will one will getDetail() again. Instead, the existing stop(programRunInfo) should call this method after it looks up the job detail.